src/cpp/rtps/messages/CDRMessage.h

```diff
@@ -350,7 +350,7 @@ inline bool skip(
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 
-#include <fastdds/rtps/messages/CDRMessage.hpp>
+#include "CDRMessage.cpp"
```

Comment:
```suggestion
```

Reply:
It fails to build without it and I think it's because in the .cpp there are all inline functions

Reply:
I think it's because the cpp is not included as a source file for the targets in the CMakeLists.txt files, you'd need to do that. I'd also remove the `inline` specifier from the functions in the .cpp